rct229/rulesets/ashrae9012019/section19/section19rule4.py

```diff
@@ -0,0 +1,321 @@
+            is_primary_rule=True,
+            list_path="ruleset_model_instances[0]",
+            required_fields={
+                "$": ["calendar"],
+                "calendar": ["day_of_week_for_january_1"],
+            },
+            data_items={
+                "day_of_week_for_january_1": (
+                    "baseline",
+                    "calendar/day_of_week_for_january_1",
+                ),
+            },
+        )
+
+    class RuleSetModelInstanceRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section19Rule4.RuleSetModelInstanceRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section19Rule4.RuleSetModelInstanceRule.ZoneRule(),
+                index_rmr="baseline",
+                list_path="buildings[*].building_segments[*].zones[*]",
+            )
+
+        def create_data(self, context, data):
+            rmi_b = context.baseline
+            schedules_b = getattr_(rmi_b, "RMI", "schedules")
+            schedule_b = {schedule["id"]: schedule for schedule in schedules_b}
+
+            assert_(rmi_b["buildings"][0], "`buildings` key doesn't exist.")
+            building_b = rmi_b["buildings"][0]
+            assert_(
+                building_b["building_segments"][0],
+                "`building_segments` key doesn't exist.",
+            )
+            building_segment_b = building_b["building_segments"][0]
+
+            is_lighting_bldg_area_defined_b = False
+            is_building_area_MF_dormitory_or_hotel_b = False
+            lighting_bldg_type_b = building_segment_b.get("lighting_building_area_type")
+            if lighting_bldg_type_b is not None:
+                is_lighting_bldg_area_defined_b = True
+            if lighting_bldg_type_b in [
+                LIGHTING_BUILDING_AREA.DORMITORY,
+                LIGHTING_BUILDING_AREA.HOTEL_MOTEL,
+                LIGHTING_BUILDING_AREA.MULTIFAMILY,
+            ]:
+                is_building_area_MF_dormitory_or_hotel_b = True
+
+            return {
+                "schedule_b": schedule_b,
+                "is_lighting_bldg_area_defined_b": is_lighting_bldg_area_defined_b,
+                "is_building_area_MF_dormitory_or_hotel_b": is_building_area_MF_dormitory_or_hotel_b,
+            }
+
+        class ZoneRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section19Rule4.RuleSetModelInstanceRule.ZoneRule, self,).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    each_rule=Section19Rule4.RuleSetModelInstanceRule.ZoneRule.SpaceRule(),
+                    index_rmr="baseline",
+                    list_path="$.spaces[*]",
+                    required_fields={
+                        "$": ["infiltration"],
+                        "infiltration": ["multiplier_schedule"],
+                    },
```

Comment:
This should not be part of the required data.
Zones could have no infiltration - for example, core zone should have no infiltration.
In that case, I would set the flag to true (which you can do it in the `create_data`)